Jenkinsfile

```diff
@@ -0,0 +1,124 @@
+import groovy.json.JsonSlurper
+// This Jenkinsfile is used by Jenkins to run the GOUpdate step of Reactome's release.
+// This step synchronizes Reactome's GO terms with Gene Ontology. 
+// It requires that the ConfirmReleaseConfigs step has been run successfully before it can be run.
+def currentRelease
+def previousRelease
+pipeline {
+	agent any
+
+	stages {
+		// This stage checks that an upstream project, ConfirmReleaseConfig, was run successfully for its last build.
+		stage('Check ConfirmReleaseConfig build succeeded'){
+			steps{
+				script{
+					// Get current release number from directory
+					currentRelease = (pwd() =~ /Releases\/(\d+)\//)[0][1];
+					previousRelease = (pwd() =~ /Releases\/(\d+)\//)[0][1].toInteger() - 1;
+					// This queries the Jenkins API to confirm that the most recent build of ConfirmReleaseConfigs was successful.
+					def configStatusUrl = httpRequest authentication: 'jenkinsKey', validResponseCodes: "${env.VALID_RESPONSE_CODES}", url: "${env.JENKINS_JOB_URL}/job/${currentRelease}/job/ConfirmReleaseConfigs/lastBuild/api/json"
+					if (configStatusUrl.getStatus() == 404) {
+						error("ConfirmReleaseConfigs has not yet been run. Please complete a successful build.")
+					} else {
+						def configStatusJson = new JsonSlurper().parseText(configStatusUrl.getContent())
+						if (configStatusJson['result'] != "SUCCESS"){
+							error("Most recent ConfirmReleaseConfigs build status: " + configStatusJson['result'] + ". Please complete a successful build.")
+						}
+					}
+				}
+			}
+		}
+		// This stage backs up the gk_central database before it is modified.
+		stage('Setup: Back up gk_central before modifications'){
+			steps{
+				script{
+					withCredentials([usernamePassword(credentialsId: 'mySQLCuratorUsernamePassword', passwordVariable: 'pass', usernameVariable: 'user')]){
+						def central_before_go_update_dump = "${env.GK_CENTRAL}_${currentRelease}_before_go_update.dump"
+						sh "mysqldump -u$user -p$pass -h${env.CURATOR_SERVER} ${env.GK_CENTRAL} > $central_before_go_update_dump"
+						sh "gzip -f $central_before_go_update_dump"
+					}
+				}
+			}
+		}
+		// Download go.obo and ec2go files from GO.
+		stage('Setup: Download go.obo and ec2go files'){
+			steps{
+				script{
+					sh "wget http://current.geneontology.org/ontology/go.obo"
+					sh "wget http://current.geneontology.org/ontology/external2go/ec2go"
+					sh "mv go.obo src/main/resources/"
+					sh "mv ec2go src/main/resources/"
```

Comment:
We might have talked about it, but I can't remember:  is the Jenkins setup taking on all the functions of the `run_GO_Update.pl` script?

Reply:
I'm not sure, I didn't look at it. Since we'd been using the Java version of it, I just cloned the repository and ran it. It told me it was missing the ec2go and go.obo files, so I added in their download. Then I just did typicaly stuff like DB backups and log compressing. Is there anything in this JF that isn't being done that should be?

Reply:
The script is under "GKB/scripts/release/go_update".  It will:

1. Clone the release-go-update repo (if necessary)
2. `git checkout` the appropriate version tag
3. `git pull` any updates
4. Copy the "go-update.properties" file, in the same directory as the perl script (i.e. "GKB/scripts/release/go_update"), to the `src/main/resources` directory fo the release-go-update repo
5. `wget` the go.obo and ec2go files
6. Do a `mvn clean compile assembly:single` to create jar file
7. Run the jar file with the path to the copied "go-update.properties" file as the first and only arg
8. Copy the "release-go-update/logs" folder into a "logs" folder in the archive directory (e.g. archive/72/logs where 'archive' is a symlink to a go_update specific archive directory).

Reply:
I'm happy to say that it seems like the Jenkins version seems to cover all those bases!